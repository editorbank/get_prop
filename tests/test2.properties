key9   =	value 9 from test2.properties  	
