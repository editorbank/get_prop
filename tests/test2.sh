set -e

source ../prop.sh

function prop(){
  prop_from $1 --env ./test.properties ./test2.properties /dev/stdin
}

function test1(){
  declare arr=(
    key1
    key2
    key3
    key4
    key5
    key6
    key7
    key8
    key9
    'key\.0'
    key10
    key11
  )
  ## now loop through the above array
  for i in "${arr[@]}"

  do
     echo "-$( prop $i )-"
  done
}

echo "key11=value from stdin" | key10="value from Environment variable"  test1 