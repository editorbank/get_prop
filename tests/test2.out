-value1-
-value2-
-value'3-
-value"4-
-value 5-
-value 6-
-value 7-
-value 8-
-value 9 from test2.properties-
-val ue .0-
-value from Environment variable-
-value from stdin-
